document/src/tests/testbytebuffer.cpp

```diff
@@ -136,251 +53,248 @@ TEST(ByteBuffer_Test, test_copy_constructor)
 
 TEST(ByteBuffer_Test, test_putGetFlip)
 {
-    ByteBuffer* newBuf=new ByteBuffer(100);
+    ByteBuffer newBuf(100);
 
     try {
-        newBuf->putInt(10);
+        newBuf.putInt(10);
         int test;
-        newBuf->flip();
+        newBuf.flip();
 
-        newBuf->getInt(test);
-        EXPECT_TRUE(test==10);
+        newBuf.getInt(test);
+        EXPECT_EQ(test, 10);
 
-        newBuf->clear();
-        newBuf->putDouble(3.35);
-        newBuf->flip();
-        EXPECT_TRUE(newBuf->getRemaining()==sizeof(double));
+        newBuf.clear();
+        newBuf.putDouble(3.35);
+        newBuf.flip();
+        EXPECT_EQ(newBuf.getRemaining(), 100);
```

Comment:
Since `getRemaining` now has different semantics than it used to, is there any other code that might be expecting the old behavior and may now change in subtle ways?